nominal/_utils.py

```diff
@@ -246,3 +139,47 @@ def update_dataclass(self: T, other: T, fields: Iterable[str]) -> None:
     """
     for field in fields:
         self.__dict__[field] = getattr(other, field)
+
+
+class FileType(NamedTuple):
+    extension: str
+    mimetype: str
+
+    @classmethod
+    def from_path(cls, path: Path | str, default_mimetype: str = "application/octect-stream") -> FileType:
+        ext = "".join(Path(path).suffixes)
+        mimetype, _encoding = mimetypes.guess_type(path)
+        if mimetype is None:
+            return cls(ext, default_mimetype)
+        return cls(ext, mimetype)
```

Comment:
nit: would this work? `cls(ext, mimetype or default_mimetype)`

Reply:
here I think it would be equivalent. By default I always explicitly check optionals with None checks since I usually want to allow for falsey values (e.g. empty string). That's not the case here but I try to maintain consistency throughout

Reply:
yeah don't mind the explicit version either, just curious - thx